org.mwc.debrief.core/src/org/mwc/debrief/core/ContextOperations/GeneratePasteRepClipboard.java

```diff
@@ -118,57 +120,36 @@ public IStatus undo(IProgressMonitor monitor, IAdaptable info)
 
   private boolean isContentImportable(final String content) {
 
-    boolean proceed=false;
-    StringTokenizer tokens = new StringTokenizer(content,"\\r?\\n");
-    int lineCount = 0;
-    while(tokens.hasMoreTokens() && lineCount<=6) {
-      String line = tokens.nextToken().trim();
+    boolean proceed=true;
+    String[] lines = content.split("\\r?\\n");
+    for(int i=0;i<lines.length;i++) {
+      String line = lines[i];
       if(line.startsWith(";") && !line.startsWith(";;")) {
         StringTokenizer lineTokens = new StringTokenizer(line);
         if(lineTokens.hasMoreTokens()) {
           String firstWord = lineTokens.nextToken();
-          switch(firstWord) {
-            case ";CIRCLE:":
-            case ";DYNAMIC_CIRCLE:":
-            case ";RECT:":
-            case ";DYNAMIC_RECT:":
-            case ";BRG:":
-            case ";DYNAMIC_POLY:":
-            case ";ELLIPSE:":
-            case ";FORMAT_FIX:":
-            case ";FORMAT_LAYER_HIDE:":
-            case ";TEXT:":
-            case ";LINE:":
-            case ";FORMAT_TRACK_NAME_AT_END:":
-            case ";NARRATIVE:":
-            case ";NARRATIVE2:":
-            case ";PERIODTEXT:":
-            case ";POLY:":
-            case ";POLYLINE:":
-            case ";SENSOR:":
-            case ";SENSOR2:":
-            case ";SENSOR3:":
-            case ";SENSOR_ARC:":
-            case ";TIMETEXT:":
-            case ";TMA_POS:":
-            case ";TMA_RB:":
-            case";VECTOR:":
-            case ";WHEEL:":
-              proceed = true;break;
-            default:proceed=false;break;
+          String regex = "^;[A-Z_]{3,40}+:$";
+          Pattern pattern = Pattern.compile(regex);
+          Matcher match = pattern.matcher(firstWord);
+          if(!match.matches()) {
+            proceed=false;
           }
         }
       }
       else {
         StringTokenizer lineTokens = new StringTokenizer(line);
         if(lineTokens.hasMoreTokens()) {
           String firstWord = lineTokens.nextToken();
-          if(firstWord.matches("\\d{6}+")) {
-            proceed=true;
+          if(!(firstWord.matches("\\d{6}+") || firstWord.matches("\\d{8}+"))) {
+            proceed=false;
           }
         }
+        
+      }
+      //we do it only for 6 lines

```

Comment:
this is really minor, but...  I think it may be tidier to use a `while` loop.  It will put all the _loop control_ logic in the same place, at the start.

I guess max_lines could Math.min(6, lines.length), then `while (i < max_lines && proceed)`

Reply:
done